### PR TITLE
Add naming conventions for instruments

### PR DESCRIPTION
**Pull request by jenshnielsen**
Triggered by #4365 let's define some standard naming conventions to avoid confusion in the future.

### CODE REVIEW

File: docs/examples/writing_drivers/Creating-Instrument-Drivers.ipynb [hunk 1]
```diff
@@ -95,6 +95,62 @@
...
+    "However, we aim to gradually update all drivers to be named as outlined above and any new driver \n",
+    "should be named in the way outlined below.\n",
+    "\n",
+    "The same rules should apply for QCoDeS-contrib-drivers with the exception that all drivers are stored in subfolders of the drivers folder. \n",
+    "\n",
+    "### Naming the Instrument class\n",
+    "A driver for an instrument with model `Model` and from the vendor `Vendor` should be stored in the file:\n",
+    "\n",
+    "```\n",
+    "qcodes\\instrument_drivers\\{Vendor}\\{Vendor{_{Model}.py \n",
```
Comment (astafan8) [inline docs/examples/writing_drivers/Creating-Instrument-Drivers.ipynb:115]:
```suggestion
    "qcodes\\instrument_drivers\\{Vendor}\\{Vendor}_{Model}.py \n",
```

File: docs/examples/writing_drivers/Creating-Instrument-Drivers.ipynb [hunk 2]
```diff
@@ -95,6 +95,62 @@
...
+    "### Naming InstrumentModule classes\n",
+    "\n",
+    "`InstrumentModule`s and `InstrumentChannel`s should be defined in the same file as the driver that they are part of. The classes should preferable be named such that it is clear\n",
+    "from the name which instrument it belongs to. E.g a hypothetical `InstrumentChannel` belonging to a Weinschel 8320 should be named `Weinschel8320Channel` or similar.\n",
+    "\n",
+    "\n",
+    "### Driver supporting multiple models.\n",
+    "\n",
+    "Often instrument vendors supply multiple instruments in a family with very similar specs only different in limits such as voltage ranges or \n",
+    "highest supported frequency. As an example have a look at the \n",
```
Comment (astafan8) [inline docs/examples/writing_drivers/Creating-Instrument-Drivers.ipynb:141]:
```suggestion
    "highest supported frequency.\n",
```

File: docs/examples/writing_drivers/Creating-Instrument-Drivers.ipynb [hunk 3]
```diff
@@ -95,6 +95,62 @@
...
+    "\n",
+    "\n",
+    "As an example have a look at the Keysight 344xxA series of digital multi meters. To implement drivers for such instruments it is preferable\n",
+    "to implement a private base class such as `_Keysight344xxA`. This class should be stored either in a `private` subfolder of the Vendor folder or\n",
+    "in a file starting with an underscore i.e. `_Keysight344xxA.py` If possible we prefer a format where `x` is used to signal the parts of the model numbers that \n",
```
Comment (astafan8) [inline docs/examples/writing_drivers/Creating-Instrument-Drivers.ipynb:146]:
```suggestion
    "in a file starting with an underscore i.e. `_Keysight344xxA.py`. If possible, we prefer a format where `x` is used to signal the parts of the model numbers that \n",
```

File: docs/examples/writing_drivers/Creating-Instrument-Drivers.ipynb [hunk 4]
```diff
@@ -95,6 +95,62 @@
...
+    "may change. Along with this class subclasses for each of the supported models should be implemented. These may either make small modifications to the baseclass as needed \n",
+    "or be empty subclasses if no modifications are needed. \n",
+    "\n",
+    "E.g. subclasses of the Keysight 344xxA driver for the specific model `34410A` should be named as `Keysight34410A` and stored in `Keysight34410A.py` \n",
```
Comment (astafan8) [inline docs/examples/writing_drivers/Creating-Instrument-Drivers.ipynb:150]:
```suggestion
    "E.g. subclasses of the Keysight 344xxA driver for the specific model `34410A` should be named as `Keysight34410A` and stored in `Keysight34410A.py`.\n",
```

File: docs/examples/writing_drivers/Creating-Instrument-Drivers.ipynb [hunk 5]
```diff
@@ -95,6 +95,62 @@
...
+    "class {Vendor}{Model}\n",
+    "     ...\n",
+    "```\n",
+    "E.g Vendor followed by Model number in CamelCase.\n",
+    "\n",
+    "Note that we use vendor names starting with upper case for both folders and file names.\n",
+    "\n",
+    "As an example the driver for the Weinschel 8320 should be stored in the file `qcodes\\instrument_drivers\\Weinschel\\Weinschel_8320.py` and the \n",
+    "class named `Weinschel8320` \n",
+    "\n",
```
Comment (astafan8) [inline docs/examples/writing_drivers/Creating-Instrument-Drivers.ipynb:131]: shall we add known short-enings for Vendors for use in instrument class names? e.g ZI is fine for Zurich Instrument, and NI is fine for National Instruments, AMI is fine for American Magnetics, etc ?

**Comment by jenshnielsen** [comment]
@trevormorgan since we are using bors we don't need to merge master back into the branch

**Comment by jenshnielsen** [comment]
bors merge

**Pull request merged**